cibuildwheel/windows.py

```diff
@@ -100,39 +100,43 @@ def shell(args, env=None, cwd=None):
 
     python_configurations = get_python_configurations(build_selector)
     for config in python_configurations:
+        # install Python
         config_python_path = get_python_path(config)
         simple_shell([nuget, "install"] + get_nuget_args(config))
-        if not os.path.exists(os.path.join(config_python_path, 'Scripts', 'pip.exe')):
-            simple_shell([os.path.join(config_python_path, 'python.exe'), get_pip_script ])
-
-        # check python & pip exist for this configuration
         assert os.path.exists(os.path.join(config_python_path, 'python.exe'))
-        assert os.path.exists(os.path.join(config_python_path, 'Scripts', 'pip.exe'))
-
-        # setup dirs
-        if os.path.exists(built_wheel_dir):
-            shutil.rmtree(built_wheel_dir)
-        os.makedirs(built_wheel_dir)
 
+        # set up PATH and environment variables for run_with_env
         env = os.environ.copy()
-        # set up environment variables for run_with_env
         env['PYTHON_VERSION'] = config.version
         env['PYTHON_ARCH'] = config.arch
         env['PATH'] = os.pathsep.join([
             config_python_path,
             os.path.join(config_python_path, 'Scripts'),
             env['PATH']
         ])
+        # update env with results from CIBW_ENVIRONMENT
         env = environment.as_dictionary(prev_environment=env)
 
         # for the logs - check we're running the right version of python
-        shell(['python', '--version'], env=env)
-        shell(['python', '-c', '"import struct; print(struct.calcsize(\'P\') * 8)\"'], env=env)
+        simple_shell(['where', 'python'], env=env)
+        simple_shell(['python', '--version'], env=env)
+        simple_shell(['python', '-c', '"import struct; print(struct.calcsize(\'P\') * 8)\"'], env=env)
+
+        # make sure pip is installed
+        if not os.path.exists(os.path.join(config_python_path, 'Scripts', 'pip.exe')):
+            simple_shell([os.path.join(config_python_path, 'python.exe'), get_pip_script], env=env)
+        assert os.path.exists(os.path.join(config_python_path, 'Scripts', 'pip.exe'))
 
         # prepare the Python environment
-        shell(['python', '-m', 'pip', 'install', '--upgrade', 'pip'], env=env)
-        shell(['pip', 'install', '--upgrade', 'setuptools'], env=env)
-        shell(['pip', 'install', 'wheel'], env=env)
+        simple_shell(['python', '-m', 'pip', 'install', '--upgrade', 'pip'], env=env)
+        simple_shell(['pip', '--version'], env=env)
+        simple_shell(['pip', 'install', '--upgrade', 'setuptools'], env=env)
+        simple_shell(['pip', 'install', 'wheel'], env=env)
```

Comment:
Same here, It feels weird no to upgrade `wheel` (and thus not having a single line: `call(['pip', 'install', '--upgrade', 'setuptools', 'wheel'], env=env))`

Any reason you know of ?

Reply:
Agreed; but I believe it's because wheel is not installed by default, so `--upgrade` wouldn't do anything.

Reply:
This seems to work perfectly on macOS at least:
```
(.venv) Matt$ pip uninstall wheel setuptools
Uninstalling wheel-0.33.6:
  Would remove:
    /Users/Matt/Dev/pypa/setuptools/.venv/bin/wheel
    /Users/Matt/Dev/pypa/setuptools/.venv/lib/python3.7/site-packages/wheel-0.33.6.dist-info/*
    /Users/Matt/Dev/pypa/setuptools/.venv/lib/python3.7/site-packages/wheel/*
Proceed (y/n)? y
  Successfully uninstalled wheel-0.33.6
Uninstalling setuptools-41.6.0:
  Would remove:
    /Users/Matt/Dev/pypa/setuptools/.venv/bin/easy_install
    /Users/Matt/Dev/pypa/setuptools/.venv/bin/easy_install-3.7
    /Users/Matt/Dev/pypa/setuptools/.venv/lib/python3.7/site-packages/easy_install.py
    /Users/Matt/Dev/pypa/setuptools/.venv/lib/python3.7/site-packages/pkg_resources/*
    /Users/Matt/Dev/pypa/setuptools/.venv/lib/python3.7/site-packages/setuptools-41.6.0.dist-info/*
    /Users/Matt/Dev/pypa/setuptools/.venv/lib/python3.7/site-packages/setuptools/*
Proceed (y/n)? y
  Successfully uninstalled setuptools-41.6.0
(.venv) Matt$ pip install --upgrade wheel setuptools
Collecting wheel
  Using cached https://files.pythonhosted.org/packages/00/83/b4a77d044e78ad1a45610eb88f745be2fd2c6d658f9798a15e384b7d57c9/wheel-0.33.6-py2.py3-none-any.whl
Collecting setuptools
  Using cached https://files.pythonhosted.org/packages/d9/de/554b6310ac87c5b921bc45634b07b11394fe63bc4cb5176f5240addf18ab/setuptools-41.6.0-py2.py3-none-any.whl
Installing collected packages: wheel, setuptools
Successfully installed setuptools-41.6.0 wheel-0.33.6
```

Reply:
Yeah, it should work; I have two explanations for this:
1) One was added afterwards.
2) @joerick somehow wanted to keep those two installations separate, for some reason?

Either way, I can change this now, and see if @joerick had a reason.

Reply:
Happy to have these both be --upgrade. Not sure why they're separate. Side tangent- I've always wanted to pin these in cibuildwheel, but it would create a maintenance burden to have to keep revving them on each release. Sure would be nice for repeatable builds though